### PR TITLE
#2024 Validation rule DWF_D_58, DWF_D_59 to update in multipart mode

### LINKED ISSUE #2024: CUSTOMER - Constraints on OEntities and OActors should be displayed in referencing elements inside the Semantic Browser view
Constraints applied on the selected element should be displayed in the Semantic Browser (left-hand part: Constraining Elements).
They are not displayed on Operational Entities. Same problem for Operational Actors (Operational Entity [Actor]).

Note that they are correctly displayed on System Actors, the System, Operational Activities and Interactions.

### PR DESCRIPTION
**Pull request by claudiasima-ro**
Fixes #2024 

Change-Id: I060d3285a5584edec28522244eb9a3e186c72543
Signed-off-by: claudiasima <claudia.sima@thalesgroup.com>

### CODE REVIEW

**Review by georgiana-ecobici — CHANGES_REQUESTED** [review]
Add issue number.

File: core/plugins/org.polarsys.capella.core.data.core.ui.quickfix/src/org/polarsys/capella/core/data/core/ui/quickfix/resolver/ConstraintLocationRule_Generator.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.ui.IMarkerResolution;
+import org.polarsys.capella.common.data.modellingcore.ModelElement;
+import org.polarsys.capella.common.tools.report.appenders.reportlogview.MarkerViewHelper;
+import org.polarsys.capella.core.data.capellacore.Constraint;
+import org.polarsys.capella.core.sirius.analysis.CsServices;
+import org.polarsys.capella.core.validation.ui.ide.quickfix.AbstractMarkerResolutionGenerator;
+
+public class ConstraintLocationRule_Generator extends AbstractMarkerResolutionGenerator {
+
```
Comment (georgiana-ecobici) [inline core/plugins/org.polarsys.capella.core.data.core.ui.quickfix/src/org/polarsys/capella/core/data/core/ui/quickfix/resolver/ConstraintLocationRule_Generator.java:28]: Add image (process).

File: core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml [hunk 2]
```diff
@@ -127,13 +127,13 @@
...
-                  {0} should constrain the Component instead of the {1}.
+                  ConstrainedElements for {0} should not be {1}.
                </message>
                <target
                      class="Constraint">
                </target>
                <description>
-                  This rule ensures that in monopart mode a constraint constrains the Component and not the Part or PartDeploymentLink.
+                  In monopart mode: This rule ensures that ConstrainedElements Value is not Part/PartDeploymentLink.
                </description>
```
Comment (georgiana-ecobici) [inline core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml:137]: This rule ensures that ConstrainedElements "Value" to "value".

File: doc/plugins/org.polarsys.capella.validation.doc/html/Validation Rules/design/well-formedness/data/ValidationRules.html [hunk 3]
```diff
@@ -759,7 +759,7 @@ <h1 id="Design_.3E_Well-formedness_.3E_Data">Design &gt; Well-formedness &gt; Da
 				<td>DWF_D_58 - Constraint Assignment </td>
 			</tr>
 			<tr>
-				<td colspan="2">This rule ensures that in monopart mode a constraint constrains the Component and not the Part or PartDeploymentLink. Exception for EAB diagrams.</td>
+				<td colspan="2">In monopart mode, this rule ensures that ConstrainedElements Value is not Part/PartDeploymentLink. Exception for EAB diagrams.</td>
```
Comment (georgiana-ecobici) [inline doc/plugins/org.polarsys.capella.validation.doc/html/Validation Rules/design/well-formedness/data/ValidationRules.html:762]: This rule ensures that ConstrainedElements "Value" to "value".

File: doc/plugins/org.polarsys.capella.validation.doc/html/Validation Rules/design/well-formedness/data/ValidationRules.mediawiki [hunk 4]
```diff
@@ -379,13 +379,16 @@ If an Association is on SystemArchitecture layer. It should not associate with a
 !|[[Image:../../../../Images/warning.gif|WARNING]]
 |DWF_D_58 - Constraint Assignment 
 |-
-| colspan="2"|This rule ensures that in monopart mode a constraint constrains the Component and not the Part or PartDeploymentLink. Exception for EAB diagrams.
+| colspan="2"|In monopart mode, this rule ensures that ConstrainedElements Value is not Part/PartDeploymentLink. Exception for EAB diagrams.
```
Comment (georgiana-ecobici) [inline doc/plugins/org.polarsys.capella.validation.doc/html/Validation Rules/design/well-formedness/data/ValidationRules.mediawiki:382]: "value"

File: core/plugins/org.polarsys.capella.core.data.core.ui.quickfix/src/org/polarsys/capella/core/data/core/ui/quickfix/resolver/ConstraintLocationRule_Generator.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  @Override
+  protected IMarkerResolution[] doGetResolutions(IMarker marker) {
+    List<IMarkerResolution> resolutions = new ArrayList<IMarkerResolution>();
+    List<EObject> modelElements = MarkerViewHelper.getModelElementsFromMarker(marker);
+    
+    if ((modelElements.size() > 0) && (modelElements.get(0) instanceof Constraint)) {
+      Constraint constraint = (Constraint) modelElements.get(0);
+      String label;
+      if (CsServices.getService().isMultipartMode((ModelElement) constraint)) {
+        label = "Move constraint under first value of ConstrainedElements";
```
Comment (georgiana-ecobici) [inline core/plugins/org.polarsys.capella.core.data.core.ui.quickfix/src/org/polarsys/capella/core/data/core/ui/quickfix/resolver/ConstraintLocationRule_Generator.java:38]: Move constraint under "the" first value of ConstrainedElements.

File: core/plugins/org.polarsys.capella.core.data.core.ui.quickfix/src/org/polarsys/capella/core/data/core/ui/quickfix/resolver/ConstraintLocationRule_Generator.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  @Override
+  protected IMarkerResolution[] doGetResolutions(IMarker marker) {
+    List<IMarkerResolution> resolutions = new ArrayList<>();
+    List<EObject> modelElements = MarkerViewHelper.getModelElementsFromMarker(marker);
+    
+    if ((!modelElements.isEmpty()) && (modelElements.get(0) instanceof Constraint)) {
+      Constraint constraint = (Constraint) modelElements.get(0);
+      String label;
+      if (CsServices.getService().isMultipartMode((ModelElement) constraint)) {
+        label = "Move constraint under the first value of ConstrainedElements";
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.core.ui.quickfix/src/org/polarsys/capella/core/data/core/ui/quickfix/resolver/ConstraintLocationRule_Generator.java:38]: the first contrained element

File: core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml [hunk 7]
```diff
@@ -127,13 +127,13 @@
                   severity="WARNING"
                   statusCode="1">
                <message>
-                  {0} should constrain the Component instead of the {1}.
+                  ConstrainedElements for {0} should not be {1}.
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml:130]: {0} should not constrain {1}

File: core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml [hunk 8]
```diff
@@ -127,13 +127,13 @@
...
                </message>
                <target
                      class="Constraint">
                </target>
                <description>
-                  This rule ensures that in monopart mode a constraint constrains the Component and not the Part or PartDeploymentLink.
+                  In monopart mode: This rule ensures that ConstrainedElements value is not Part/PartDeploymentLink.
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml:136]: that a constraint is not constraining a Part nor PartDeploymentLink

File: core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml [hunk 9]
```diff
@@ -152,8 +152,11 @@
                      class="Constraint">
                </target>
                <description>
-                  This rule ensures that a constraint constraining elements is not stored unexpectedly. 
-                  (A constraint shall not be stored under a part if not constraining parts or under a deployment link if not constraining deployment links.)
+                  In monopart mode: This rule ensures that a constraint is not stored under and have ConstrainedElements values as a Part/PartDeploymentLink.
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml:155]: and is contraining a Part or a PartDeploymentLink

File: core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml [hunk 10]
```diff
@@ -152,8 +152,11 @@
...
+                  Exception : In EPBS Layer constraint's can be stored under Part
+
+                  In multipart mode : This rule ensures that a constraint is not stored under PartDeploymentlink.
+                  Exception : If first ConstraintElements value is container PartDeploymentlink or empty ConstraintElements value
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.data.core.validation/plugin.xml:159]: If a constraint is not constraining any element or if its stored under and constraining only a PartDeploymentLink

File: doc/plugins/org.polarsys.capella.validation.doc/html/Validation Rules/design/well-formedness/data/ValidationRules.html [hunk 11]
```diff
@@ -759,7 +759,7 @@ <h1 id="Design_.3E_Well-formedness_.3E_Data">Design &gt; Well-formedness &gt; Da
 				<td>DWF_D_58 - Constraint Assignment </td>
 			</tr>
 			<tr>
-				<td colspan="2">This rule ensures that in monopart mode a constraint constrains the Component and not the Part or PartDeploymentLink. Exception for EAB diagrams.</td>
+				<td colspan="2">In monopart mode, this rule ensures that ConstrainedElements value is not Part/PartDeploymentLink. Exception for EAB diagrams.</td>
```
Comment (pdulth) [inline doc/plugins/org.polarsys.capella.validation.doc/html/Validation Rules/design/well-formedness/data/ValidationRules.html:762]: that a constrain is not constraining a Part nor a PartDeploymentLink

**Review by pdulth — CHANGES_REQUESTED** [review]
Hi, 
the "ConstrainedElements value" is not really user focused.
otherwise seems ok

**Comment by pdulth** [comment]
Hmm, i see in story that it is expected.. 🤨 I guess its ok then

**Review by georgiana-ecobici — APPROVED** [review]

**Review by pdulth — APPROVED** [review]

**Pull request merged**